### PR TITLE
Add multiarch support for go runner image

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <udit.gaurav@mayadata.io>

#### Issue:
- fixes: https://github.com/litmuschaos/litmus/issues/1550

#### Details:

- It includes muti-arch support for litmus `go-runner` image which includes `linux/adm64` and `linux/arm64` support. We also need a script for cross-compilation of the go binaries for different experiments and generalizing the Dockerfile to have support for different arch which has been added in this PR.

- It also includes some Makefile enhancements like golint, govet and gofmt check also a travis security check is added. The Travis version update and docker version updation are also being added in this PR.

### CODE REVIEW

File: build/litmus-go/Dockerfile [hunk 1]
```diff
@@ -14,37 +16,34 @@ ENV STRESS_NG_VERSION ${STRESS_NG_VERSION:-0.10.10}
...
-RUN apt-get install stress
-
 # make static version
 WORKDIR /code/stress-ng
 RUN STATIC=1 make
 
 FROM alpine:latest
 
-LABEL maintainer="LitmusChaos"
+ARG TARGETARCH
```
Comment (xUnholy) [inline build/litmus-go/Dockerfile:25]: In the build stage of the Dockerfile we should use the native `buildx` ARG which doesn't require us to add additional `--build-args` ourselves.

We should use `TARGETPLATFORM`. What this allows is for us to also get and build for the OS/ARCH/ARMv using the following example

```
 export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) && \
    export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2) && \
    export GOARM=$(echo ${TARGETPLATFORM} | cut -d / -f3 | cut -c2-) && \
    # build something here
```

Comment (xUnholy) [reply build/litmus-go/Dockerfile:25]: We should also consider restructuring this Dockerfile for overall enhancements and improvements, I'm also not sure about `TARGETARCH` did some quick reading and looking at your build commands and it doesn't look like it's being passed, how does this work? I didn't read it in the official Docs, I only saw `TARGETPLATFORM` so I'm keen to learn how this is picking up the arch currently.

Comment (uditgaurav) [reply build/litmus-go/Dockerfile:25]: Thanks, @xUnholy for reviewing it.
The Docker predefines a set of `ARG` variable according to the `--platform` we pass in buildx. This is only available when using the BuildKit backend. Here the `TARGETARCH` is the architecture component of `TARGETPLATFORM` . 

ref: https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

Comment (xUnholy) [reply build/litmus-go/Dockerfile]: @uditgaurav awesome this is new that is fantastic! Thanks for the link 👍 I'm happy with this improvement <3

**Review by xUnholy — DISMISSED** [review]
LGTM

**Review by ksatchit — APPROVED** [review]
LGTM

**Pull request merged**